src/Files/ViewModels/NavToolbarViewModel.cs

```diff
@@ -933,26 +934,9 @@ public async Task CheckPathInput(string currentInput, string currentSelectedPath
                                 ? CommonPaths.HomePath
                                 : shellPage.FilesystemViewModel.WorkingDirectory;
 
-                            // Launch terminal application if possible
-                            foreach (var terminal in App.TerminalController.Model.Terminals)
+                            if (await LaunchApplicationFromPath(currentInput, workingDir))
                             {
-                                if (terminal.Path.Equals(currentInput, StringComparison.OrdinalIgnoreCase)
-                                    || terminal.Path.Equals(currentInput + ".exe", StringComparison.OrdinalIgnoreCase) || terminal.Name.Equals(currentInput, StringComparison.OrdinalIgnoreCase))
-                                {
-                                    var connection = await AppServiceConnectionHelper.Instance;
-                                    if (connection != null)
-                                    {
-                                        var value = new ValueSet()
-                                        {
-                                            { "Arguments", "LaunchApp" },
-                                            { "WorkingDirectory", workingDir },
-                                            { "Application", terminal.Path },
-                                            { "Parameters", string.Format(terminal.Arguments, workingDir) }
-                                        };
```

Comment:
Note: I've also removed the special handling for launching terminal apps, LaunchApplicationFromPath can handle launching terminals as well.

Reply:
Does `wt` open in the correct path for you?

Reply:
It does, if it does not for you we can bring the code above back

Reply:
~It does~. Noope I was trying from the toolbar, apparently we do need to pass the "Parameters" from the terminals profile.

Reply:
Restored

Reply:
Thanks!